src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,75 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
-    private static final Injector injector = new Injector();
+    private static final Injector INJECTOR = new Injector();
+    private static final Map<Class<?>, Object> INSTANCES = new HashMap<>();
```

Comment:
camelCase. These are not constants.

Reply:
Bogdan in the video uses "private static final Injector injector = new Injector();" I'm confused. It looks like a constant and should be in the upper case. Now I see that "static final" is not necessarily a constant. Leonid, can you tell me what the rule is here?

Reply:
https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names
If we can change inner state of the object - it is NOT constant
![image](https://user-images.githubusercontent.com/20326858/224999800-6e5a80d4-d3ef-45c2-8d88-1ccc24235064.png)